brainzutils/musicbrainz_db/tests/test_artist.py

```diff
@@ -1,26 +1,21 @@
-from unittest import TestCase
-from mock import MagicMock
+import pytest
+
 from brainzutils.musicbrainz_db.test_data import artist_linkin_park, artist_jay_z
 from brainzutils.musicbrainz_db.unknown_entities import unknown_artist
 from brainzutils.musicbrainz_db import artist as mb_artist
 
 
-class ArtistTestCase(TestCase):
-
-    def setUp(self):
-        mb_artist.mb_session = MagicMock()
-        self.mock_db = mb_artist.mb_session.return_value.__enter__.return_value
-        self.artist_query = self.mock_db.query.return_value.options.return_value.filter.return_value.all
+@pytest.mark.database
+class TestArtist:
 
-    def test_get_by_id(self):
-        self.artist_query.return_value = [artist_linkin_park]
+    def test_get_by_id(self, engine):
```

Comment:
this is how you specify that a test requires a fixture. By marking that this test needs the sqlalchemy engine, it'll connect to the database before running this test. Fixtures are nicer than setUp methods, because you can selectively apply them to only the methods that you need